### ISSUE TITLE
Time zone issue when using timePicker

### ISSUE THREAD

**Issue by kristiansu** [hackiftekhar/IQDropDownTextField#54]
Hi!

I'm experiencing the following issue:

My device is set to CEST time zone (UTC +2), but when i select time from time picker drop down, time is being stored in UTC +1 time zone. Now, i'm not 100% percent sure this issue occurs because there's something wrong with the library, but is there a chance that there might be something wrong? Could you look into it?

Best regards!

**Comment by hackiftekhar** [comment]
You could try changing timezone of `dateFormatter` or `timeFormatter` property. Let me know if this fixed your issue.

**Comment by kristiansu** [comment]
UPDATE:
When using Time picker, library returns date in form of e.g. 01-01-2000 10:33:22 , and since the date is in  january there is no daylight saving time and therefore it converts to CET (UTC+1) instead of CEST(UTC +2)

**Comment by serg94** [comment]
+

So what's the solution?

**Comment by serg94** [comment]
```Swift
            let x = self.TEXT_FIELD.text!
            let f = NSDateFormatter()
            f.timeZone = NSTimeZone.systemTimeZone()
            f.dateFormat = "hh:mm a"
            var d = f.dateFromString(x)
            let offset = Double(NSTimeZone.localTimeZone().secondsFromGMT)
            d = d?.dateByAddingTimeInterval(-offset)
```

this is my hot fix, just in case, may be it can help to fix in general.

**Comment by lecksfrawen** [comment]
Following @hackiftekhar advice I used dateFormatter, but by setting a locale for "es-MX" I can't pick any date. It scrolls back to the minimumDate after any date is picked. Without setting locale I have no issues though.

I ended removing the locale configuration and changed my simulator/device region/language settings to Spanish/Mexico and I get the picker in Spanish as I desired. Just a heads up.
